e2e/deployers/disapp.go

```diff
@@ -21,12 +22,35 @@ func (d DiscoveredApp) GetNamespace(ctx types.TestContext) string {
 }
 
 // Deploy creates a workload on the first managed cluster.
+// nolint:funlen,cyclop
 func (d DiscoveredApp) Deploy(ctx types.TestContext) error {
 	log := ctx.Logger()
 	appNamespace := ctx.AppNamespace()
 
-	// Deploys the application on the first DR cluster (c1).
-	cluster := ctx.Env().C1
+	var cluster types.Cluster
+
+	statuses, err := ctx.Workload().Status(ctx)
+	if err != nil {
+		return err
+	}
+
+	switch len(statuses) {
+	case 0:
+		cluster = ctx.Env().C1
+		log.Debugf("Application \"%s/%s\" not found, deploying on cluster %q",
+			appNamespace, ctx.Workload().GetAppName(), cluster.Name)
+	case 1:
+		log.Debugf("Application \"%s/%s\" found in cluster %q with status %q",
+			appNamespace, ctx.Workload().GetAppName(), statuses[0].ClusterName, statuses[0].Status)
+
+		cluster, err = ctx.Env().GetCluster(statuses[0].ClusterName)
+		if err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("application \"%s/%s\" found on multiple clusters: %+v",
+			appNamespace, ctx.Workload().GetAppName(), statuses)
+	}
```

Comment:
Instead of adding //nolint, we can put this block in a helper like:

```go
chooseDeployCluster(ctx types.TestContext) (types.Cluster, error)
```

This will minimizes the change in this function, and make the code more clear.